File: ArkhamAsylum.Lib/Models/Room.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace ArkhamAsylum.Lib.Models
{
    public class Room : Entity
    {
        public int Number { get; set; }

        public Guid FloorId { get; set; }

        public Guid AreaId { get; set; }

        [JsonIgnore]
        public Floor Floor { get; set; }

        [JsonIgnore]
        public Area Area { get; set; }

        [JsonIgnore]
        public ICollection<Bed> Beds { get; set; }

        [JsonIgnore]
        public ICollection<NurseRoomAssignation> NurseRoomAssignations { get; set; }
    }
}

File: ArkhamAsylum.Web/wwwroot/Scripts/Models/Room.js
class Room extends Entity
{
    constructor(json)
    {
        super(json);
        if (json)
        {
            this.Number = json.number;
            this.FloorId = json.floorId;
            this.AreaId = json.areaId;
        }
        else
        {
            this.Number = 0;
            this.FloorId = "00000000-0000-0000-0000-000000000000";
            this.AreaId = "00000000-0000-0000-0000-000000000000";
        }
    }
}